File: python/problem7.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        s = str(abs(x));
        if x >= 0:
            r = int(s[::-1]);
        else:
            r = -int(s[::-1]);
        if r > 2**31 - 1 or r < -2**31:
            return 0;
        else:
            return r;

File: Java/problem7.java
public class Solution
{
    public int reverse(int x) 
    {
        int temp = x;
        int r, result = 0, newResult = 0;
        while(temp != 0)
        {
            r = temp % 10;
            temp = temp / 10;
            newResult = result * 10 + r;
            if((newResult - r)/10 != result)        //溢出
                return 0;
            result = newResult;
        }
        return result;
    }
}